tests/Avalonia.Controls.UnitTests/AutoCompleteBoxTests.cs

```diff
@@ -1012,25 +1012,25 @@ private TextBox GetTextBox(AutoCompleteBox control)
         }
         private IControlTemplate CreateTemplate()
         {
-            return new FuncControlTemplate<AutoCompleteBox>(control =>
+            return new FuncControlTemplate<AutoCompleteBox>((control, scope) =>
             {
                 var textBox =
                     new TextBox
                     {
                         Name = "PART_TextBox"
-                    };
+                    }.RegisterInNameScope(scope);
```

Comment:
Would it make sense to have an overload of `FuncControlTemplate` which automatically registers named controls to save on the noise of manually calling `RegisterInNameScope` for each one?

Reply:
I'm going to change `ITemplate` to return the content *and* name scope. That will avoid attached property pollution. I'll keep the attached properties for XAML roots for `this.FindControl<T>("name")`.